packages/core/src/use-cases/uc-type-designer.ts

```diff
@@ -0,0 +1,33 @@
+import type { SCDQueries } from "../scd/scd-query"
+
+/** 
+ * The name is temporary, rename it if you have a better one
+ * UC = Use Case
+ */
+export class UCTypeDesigner {
+
+	constructor(
+		private readonly scdQueries: SCDQueries,
+	){}
+
+	public findAllLogicalDevices(){
+		return this.scdQueries.searchLDeviceType();
+	}
+
+	public findAllBays(){
+		return this.scdQueries.searchBayType();
+	}
+
+	public findAllIEDs(){
+		return this.scdQueries.searchIEDType();
+	}
+
+	public findAllVoltageLevels(){
+		return this.scdQueries.searchVoltageLevelType();
+	}
+
+	public findAllSubstations(){
+		return this.scdQueries.searchSubstationType();
+	}
+
+}
```

Comment:
In the UC Class we can combine and prepare all the date we need, so I suggest moving this code here

```
        logicalDevices: ucci.findAllLogicalDevices(),
        bays: ucci.findAllBays(),
        ieds: ucci.findAllIEDs(),
        voltageLevels: ucci.findAllVoltageLevels(),
        substations: ucci.findAllSubstations(),
```

and return a single object with all the needed information. And we should limit the root of the querries to `<DataTypeTemplates>` so we dont have to scan the entire File for every Type.

Reply:
Implemented as commented by @clepski.